sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/postgreSQL/PostgreCastBase.scala

```diff
@@ -16,19 +16,29 @@
  */
 package org.apache.spark.sql.catalyst.expressions.postgreSQL
 
+import java.time.ZoneId
+
+import org.apache.spark.sql.AnalysisException
 import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
 import org.apache.spark.sql.catalyst.expressions.{CastBase, Expression, TimeZoneAwareExpression}
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, JavaCode}
 import org.apache.spark.sql.catalyst.expressions.codegen.Block._
+import org.apache.spark.sql.catalyst.util.DateTimeUtils
 import org.apache.spark.sql.catalyst.util.postgreSQL.StringUtils
-import org.apache.spark.sql.types._
+import org.apache.spark.sql.types.{BooleanType, DataType, DateType, IntegerType, NullType, StringType, TimestampType}
 import org.apache.spark.unsafe.types.UTF8String
 
-case class PostgreCastToBoolean(child: Expression, timeZoneId: Option[String])
-  extends CastBase {
+abstract class PostgreCastBase extends CastBase{
```

Comment:
nit: `CastBase{` => `CastBase {`. Can you check again more carefully?

Reply:
Done